tests/unit/dojson/test_dojson_hep.py

```diff
@@ -733,6 +733,52 @@ def test_authors_from_100__a_u_x_w_y_z_with_malformed_x():
     assert expected == result['authors']
 
 
+def test_authors_from_100__a_double_m_double_u_w_y_z():
+    snippet = (
+        '<datafield tag="100" ind1=" " ind2=" ">'
+        '  <subfield code="a">Puy, Denis</subfield>'
+        '  <subfield code="m">puy@tsmi19.sissa.it</subfield>'
+        '  <subfield code="m">puy@mesioa.obspm.fr</subfield>'
+        '  <subfield code="u">SISSA, Trieste</subfield>'
+        '  <subfield code="u">Meudon Observ.</subfield>'
+        '  <subfield code="w">D.Puy.2</subfield>'
+        '  <subfield code="y">0</subfield>'
+        '  <subfield code="z">903393</subfield>'
+        '</datafield>'
+    )  # record/413614/export/xme
+
+    expected = [
+        {
+            'affiliations': [
+                {
+                    'record': {
+                        '$ref': 'http://localhost:5000/api/institutions/903393',
+                    },  # XXX: wrong, fix incoming.
```

Comment:
Just noticed that the thing I did with `zip_longest` is completely wrong, and @mihaibivol was right in zipping only when the `len`s match.